src/student.py

```diff
@@ -7,22 +7,65 @@ def __init__(self, student_id, subjects_and_classes, subject_targets, subject_gi
         self.buddies = buddies   #Dictionary where for each subject (key) there is a list of ups ordered by their priority
     
     def add_subject_and_class(self, subject_name, class_number):
-        self.subjects_and_classes[subject_name] = class_number
+        if len(self.subjects_and_classes) < 7:
+            self.subjects_and_classes[subject_name] = class_number
 
     def add_subject_target(self, subject_name, class_number):
-        self.subject_targets[subject_name].append(class_number)
+        if subject_name in self.subjects_and_classes:
+            if subject_name not in self.subject_targets:
+                self.subject_targets[subject_name] = [class_number]
+            else:
+                self.subject_targets[subject_name].append(class_number)
+
```

Comment:
Just for good code practice and future utility, it would be nice having the functions returning 0 upon success and 1 otherwise.

Reply:
Already done!